pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go

```diff
@@ -66,9 +78,9 @@ leaderElection:
   leaderElect: true`
 )
 
-// KubeScheduler contains function for a kube-scheduler deployer.
+// KubeScheduler contains functions for a kube-scheduler deployer.
 type KubeScheduler interface {
-	component.DeployWaiter
+	component.ObservableDeployWaiter
```

Comment:
See the comment above, I would declare implementation of both interfaces separately here, something like this:
```suggestion
	component.DeployWaiter
	component.MonitoringComponent
```